aggregator/operator_registrations_inmemory.go

```diff
@@ -0,0 +1,190 @@
+package aggregator
+
+import (
+	"context"
+	"sync"
+
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/avsregistry"
+	"github.com/Layr-Labs/eigensdk-go/crypto/bls"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/Layr-Labs/eigensdk-go/services/operatorpubkeys"
+	"github.com/Layr-Labs/eigensdk-go/types"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type OperatorRegistrationsService interface {
+	operatorpubkeys.OperatorPubkeysService
+
+	GetOperatorPubkeysById(ctx context.Context, operatorId types.OperatorId) (operatorPubkeys types.OperatorPubkeys, operatorFound bool)
+}
+
+type OperatorRegistrationsServiceInMemory struct {
+	avsRegistrySubscriber avsregistry.AvsRegistrySubscriber
+	avsRegistryReader     avsregistry.AvsRegistryReader
+	logger                logging.Logger
+	queryByAddrC          chan<- queryByAddr
+	queryByIdC            chan<- queryById
+}
+
+type queryByAddr struct {
+	operatorAddr common.Address
+	// channel through which to receive the response (operator pubkeys)
+	respC chan<- resp
+}
+type queryById struct {
+	operatorId types.OperatorId
+	// channel through which to receive the response (operator pubkeys)
+	respC chan<- resp
+}
+
+type resp struct {
```

Comment:
This is probably from eigensdk itself, but maybe `queryResp` would already be an improvement

Reply:
Keeping it like this for consistency with `eigen-sdk`